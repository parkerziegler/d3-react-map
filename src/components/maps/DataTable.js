import * as React from 'react'
import { connect } from 'react-redux';
import '../../App.css'
import * as _ from 'lodash';

class DataTable extends React.Component {

	constructor() {
		super();
		this.state = { activeTab: 'highest', isHoverOnHighest: false, isHoverOnLowest: false };
		this.onMouseEnterHandler = this.onMouseEnterHandler.bind(this);
		this.onMouseLeaveHandler = this.onMouseLeaveHandler.bind(this);
		this.onClickHandler = this.onClickHandler.bind(this);
	}

	onMouseEnterHandler(event) {

		if (event.target.id === 'highest') {
			this.setState({
				...this.state,
				isHoverOnHighest: true,
				isHoverOnLowest: false
			});
		} else if (event.target.id === 'lowest') {
			this.setState({
				...this.state,
				isHoverOnHighest: false,
				isHoverOnLowest: true
			});
		} else {
			return;
		}
	}

	onMouseLeaveHandler(event) {

			this.setState({
				...this.state,
				isHoverOnHighest: false,
				isHoverOnLowest: false
			});
	}

	onClickHandler(event) {
		if (event.target.id === 'highest') {
			this.setState({
				...this.state,
				activeTab: 'highest'
			});
		} else if (event.target.id === 'lowest') {
			this.setState({
				...this.state,
				activeTab: 'lowest'
			});
		} else {
			return;
		}
	}

	render() {

		let active = {background: '#4292C6', color: '#FFFFFF', border: '1px solid #6BAED6'};
		let inactive = {background: '#FFFFFF', color: '#4292C6', border: '1px solid #4292C6'};

		let highest = this.state.isHoverOnHighest || this.state.activeTab === 'highest' ? active : inactive;
		let lowest = this.state.isHoverOnLowest || this.state.activeTab === 'lowest' ? active : inactive;

		let ascDesc = this.state.activeTab === 'highest' ? ['desc'] : ['asc'];

		let states = this.props.maps.data.objects ?
				_.slice(_.orderBy(this.props.maps.data.objects.states.geometries, ['properties.shootingsPerCapita'], ascDesc), 0, 5) : null;

		let stateNames = _.map(states, (state, i) => {
			return <div className='row' key={i}>{state.properties.stateName}</div>;
		});

		let shootingsPerMillion = _.map(states, (state, i) => {
			return <div className='row' key={i}>{_.round(state.properties.shootingsPerCapita * 1000000, 2)}</div>;
		});

		return (
			<div className='data-table'>
				<div className='table-button-container'>
					<div className='button' id='highest' style={highest} onMouseEnter={this.onMouseEnterHandler} onMouseLeave={this.onMouseLeaveHandler} onClick={this.onClickHandler}>Highest</div>
					<div className='button' id='lowest' style={lowest} onMouseEnter={this.onMouseEnterHandler} onMouseLeave={this.onMouseLeaveHandler}
					onClick={this.onClickHandler}>Lowest</div>
				</div>
				<div className='table'>
					<div>
						<div className='table-header'>State</div>
						{stateNames}
					</div>
					<div>
						<div className='table-header'>Shootings Per Million</div>
						{shootingsPerMillion}
					</div>
				</div>
			</div>
		);
	}
}

function mapStateToProps (state, ownProps) {
  return {
    maps: state.rootReducer
  }
}

export default connect(mapStateToProps)(DataTable);
