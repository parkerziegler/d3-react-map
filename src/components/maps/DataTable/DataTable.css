/** A constants SCSS partial for storing common variables and mixins */
@import url("https://fonts.googleapis.com/css?family=Roboto");
.data-table {
  display: flex;
  flex-direction: column;
  flex: 0 0 auto; }
  .data-table .table-button-container {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    width: 60%; }
    .data-table .table-button-container .button {
      font-size: 10px;
      font-family: "HelveticaNeue", "Helvetica", sans-serif;
      width: 40%;
      height: auto;
      background: #50505E;
      flex: 0 0 auto;
      padding: 5px 5px 5px 5px;
      text-align: center;
      cursor: pointer; }
      @media screen and (min-width: 480px) {
        .data-table .table-button-container .button {
          font-size: 12px; } }
      @media screen and (min-width: 768px) {
        .data-table .table-button-container .button {
          font-size: 15px; } }
      @media screen and (min-width: 1024px) {
        .data-table .table-button-container .button {
          font-size: 18px; } }
    .data-table .table-button-container div:first-child {
      border-top-left-radius: 20px;
      border-bottom-left-radius: 20px; }
    .data-table .table-button-container div:last-child {
      border-top-right-radius: 20px;
      border-bottom-right-radius: 20px; }
  .data-table .table {
    display: flex;
    flex-direction: row;
    margin-top: 10px; }
    .data-table .table div {
      display: flex;
      flex-direction: column;
      width: 80%; }
      .data-table .table div .table-header {
        font: 700 13px "HelveticaNeue", "Helvetica", sans-serif;
        padding: 5px 0 5px 0; }
      .data-table .table div .row {
        font: 400 12px "Roboto", sans-serif;
        padding: 2px 0 2px 0; }
