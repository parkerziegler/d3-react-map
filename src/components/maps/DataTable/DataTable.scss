@import '../../../stylesheets/constants.scss';

$button-font-sizes: (
    null: 10px,
    mobile: 12px,
    tablet: 15px,
    desktop: 18px
);

.data-table {
    @include flex-column;

    .table-button-container {
        @include flex-row;
        justify-content: space-around;
        width: 60%;

        .button {
            @include font-size($button-font-sizes);
            font-family: $helvetica;
            width: 40%;
            height: auto;
            background: #50505E;
            flex: 0 0 auto;
            padding: 5px 5px 5px 5px;
            text-align: center;
            cursor: pointer;
        }

        div:first-child {
            border-top-left-radius: 20px;
            border-bottom-left-radius: 20px;
        }

        div:last-child {
            border-top-right-radius: 20px;
            border-bottom-right-radius: 20px;
        }
    }

    .table {
        @include flex-row;
        margin-top: 10px;

        div {
            @include flex-column;
            width: 80%;

            .table-header {
                font: 700 13px $helvetica;
                padding: 5px 0 5px 0;
            }

            .row {
                font: 400 12px $roboto;
                padding: 2px 0 2px 0;
            }
        }
    }
}