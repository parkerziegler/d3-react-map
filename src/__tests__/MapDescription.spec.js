import MapDescription from '../components/maps/MapDescription/MapDescription';
import DataTable from '../components/maps/DataTable/DataTable';
import { store } from '../index';

describe('<MapDescription />', () => {
  it('renders properly', () => {
    const wrapper = shallow(
      <MapDescription mapType="choropleth" store={store} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders initialState properly', () => {
    const wrapper = shallow(
      <MapDescription mapType="choropleth" store={store} />
    ).dive();
    expect(wrapper.find('.state-name').text()).toEqual('New Mexico');

    const subheaders = wrapper.find('.inset-subheader');
    expect(subheaders.first().text()).toEqual('');
    expect(subheaders.last().text()).toEqual('20.66 shootings per million');
    expect(wrapper.find(DataTable)).toHaveLength(1);
  });

  it('renders different statistics based on the mapType', () => {
    const wrapper = shallow(
      <MapDescription mapType="proportional" store={store} />
    ).dive();
    expect(wrapper.find('.state-name').text()).toEqual('New Mexico');

    const subheaders = wrapper.find('.inset-subheader');
    expect(subheaders.first().text()).toEqual('');
    expect(subheaders.last().text()).toEqual('43 shootings');
    expect(wrapper.find(DataTable)).toHaveLength(1);
  });
});
