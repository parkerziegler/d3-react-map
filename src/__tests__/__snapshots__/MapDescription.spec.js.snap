// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapDescription /> renders properly 1`] = `
<MapDescription
  dispatch={[Function]}
  mapType="choropleth"
  maps={
    Object {
      "activeState": Object {
        "shootings": 43,
        "shootingsPerMillion": 20.66,
        "stateName": "New Mexico",
      },
      "censusData": Array [],
      "fetchingData": true,
      "geoData": Array [],
      "shootingsByDate": Array [],
      "shootingsData": Array [],
      "temporalFilter": "100",
    }
  }
  router={
    Object {
      "basename": "/d3-react-map",
      "hash": "",
      "key": undefined,
      "params": Object {},
      "pathname": "/",
      "query": Object {},
      "queue": Array [],
      "result": Object {
        "index": 0,
        "title": "Intro",
      },
      "route": "/",
      "routes": Object {
        "/": Object {
          "index": 0,
          "title": "Intro",
        },
        "/percapita": Object {
          "descSubtitle": "By State",
          "descTitle": "Shootings Per Million",
          "hasChildren": true,
          "index": 2,
          "jsx": <div
            className="text"
    >
            Of all 50 states, 
            <b>
                    New Mexico
            </b>
             had the greatest number of police shootings per million residents between 2015 and 2016. Western states in general demonstrated higher rates than the eastern and northern United States, with the 
            <b>
                    District of Columbia
            </b>
             being an exception. Four of the five lowest rates were found in the northeastern United States. 
            <br />
            <br />
            Hover over a state to obtain the number of shootings per million.
    </div>,
          "title": "Shootings Per Million By State",
          "type": "map",
        },
        "/percapita/asian": Object {
          "childIndex": 2,
          "descSubtitle": "Asian",
          "descTitle": "Shootings Per Million",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            On a per capita basis, Asians were the least likely of all races to be killed by police between 2015 and 2016. In fact,
             
            <b>
                    a majority of states (35)
            </b>
             had no shootings of Asians during this time period. 
            <b>
                    Colorado
            </b>
            , 
            <b>
                    Louisiana
            </b>
            , and 
            <b>
                    Hawaii
            </b>
             were major outliers to this trend, demonstrating the highest rates of shootings of Asians.
            <br />
            <br />
            Hover over a state to obtain the number of shootings per million.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Shootings Per Million",
            "hasChildren": true,
            "index": 2,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        New Mexico
              </b>
               had the greatest number of police shootings per million residents between 2015 and 2016. Western states in general demonstrated higher rates than the eastern and northern United States, with the 
              <b>
                        District of Columbia
              </b>
               being an exception. Four of the five lowest rates were found in the northeastern United States. 
              <br />
              <br />
              Hover over a state to obtain the number of shootings per million.
    </div>,
            "route": "/percapita",
            "title": "Shootings Per Million By State",
            "type": "map",
          },
          "title": "Shootings Per Million By Race Asian",
          "type": "map",
        },
        "/percapita/black": Object {
          "childIndex": 0,
          "descSubtitle": "African American",
          "descTitle": "Shootings Per Million",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            Per million African American residents, 
            <b>
                    Oklahoma
            </b>
             had the highest rate of police shootings at 
            <b>
                    68.21
            </b>
            . 
            <b>
                    West Virginia
            </b>
             and
             
            <b>
                    Nebraska
            </b>
             had the second and third highest rates at 
            <b>
                    48.89
            </b>
             and
             
            <b>
                    43.45
            </b>
             respectively. Only 
            <b>
                    10
            </b>
             states had no shootings of African Americans in this time period. 
            <br />
            <br />
            Hover over a state to obtain the number of shootings per million.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Shootings Per Million",
            "hasChildren": true,
            "index": 2,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        New Mexico
              </b>
               had the greatest number of police shootings per million residents between 2015 and 2016. Western states in general demonstrated higher rates than the eastern and northern United States, with the 
              <b>
                        District of Columbia
              </b>
               being an exception. Four of the five lowest rates were found in the northeastern United States. 
              <br />
              <br />
              Hover over a state to obtain the number of shootings per million.
    </div>,
            "route": "/percapita",
            "title": "Shootings Per Million By State",
            "type": "map",
          },
          "title": "Shootings Per Million By Race African American",
          "type": "map",
        },
        "/percapita/latino": Object {
          "childIndex": 1,
          "descSubtitle": "Latino",
          "descTitle": "Shootings Per Million",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            The western United States demonstrated a much higher incidence of police involved shootings of Latinos than other parts of the country. Southwest border states like 
            <b>
                    New Mexico
            </b>
             (
            <b>
                    24.23
            </b>
             shootings per million Latino residents) and intermountain West states like 
            <b>
                    Idaho
            </b>
             (
            <b>
                    20.91
            </b>
            ) had generally higher rates than eastern states. 
            <br />
            <br />
            Hover over a state to obtain the number of shootings per million.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Shootings Per Million",
            "hasChildren": true,
            "index": 2,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        New Mexico
              </b>
               had the greatest number of police shootings per million residents between 2015 and 2016. Western states in general demonstrated higher rates than the eastern and northern United States, with the 
              <b>
                        District of Columbia
              </b>
               being an exception. Four of the five lowest rates were found in the northeastern United States. 
              <br />
              <br />
              Hover over a state to obtain the number of shootings per million.
    </div>,
            "route": "/percapita",
            "title": "Shootings Per Million By State",
            "type": "map",
          },
          "title": "Shootings Per Million By Race Latino",
          "type": "map",
        },
        "/percapita/nativeamerican": Object {
          "childIndex": 3,
          "descSubtitle": "Native American",
          "descTitle": "Shootings Per Million",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            <b>
                    Native americans are killed by police at a higher rate than any other race in the country.
            </b>
             
            While Midwestern states with larger indigenous populations demonstrate consistently inflated rates, northwestern states like 
            <b>
                    Alaska
            </b>
             and
             
            <b>
                    Washington
            </b>
             retain many of the highest rates. 
            <b>
                    Vermont
            </b>
             is an extreme outlier in the dataset, its rate of shootings being the result of a small indigenous population.
            <br />
            <br />
            Hover over a state to obtain the number of shootings per million.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Shootings Per Million",
            "hasChildren": true,
            "index": 2,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        New Mexico
              </b>
               had the greatest number of police shootings per million residents between 2015 and 2016. Western states in general demonstrated higher rates than the eastern and northern United States, with the 
              <b>
                        District of Columbia
              </b>
               being an exception. Four of the five lowest rates were found in the northeastern United States. 
              <br />
              <br />
              Hover over a state to obtain the number of shootings per million.
    </div>,
            "route": "/percapita",
            "title": "Shootings Per Million By State",
            "type": "map",
          },
          "title": "Shootings Per Million By Race Native American",
          "type": "map",
        },
        "/percapita/white": Object {
          "childIndex": 4,
          "descSubtitle": "White",
          "descTitle": "Shootings Per Million",
          "hasNextSibling": false,
          "isLastChildRoute": true,
          "jsx": <div
            className="text"
    >
            Shootings of whites by police closely mirrored the geography of the nation as a whole. Western states like 
            <b>
                    Oklahoma
            </b>
            , 
            <b>
                    Wyoming
            </b>
            , and
             
            <b>
                    New Mexico
            </b>
             demonstrated the highest rates while the northeastern United States had consistently low rates. Of all races, whites were the second least likely to be shot by the police. Perhaps most startingly, no whites were killed in the 
            <b>
                    District of Columbia
            </b>
             during this time period, meaning the city's high rate of police shootings were exclusively of people of color.
            <br />
            <br />
            Hover over a state to obtain the number of shootings per million.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Shootings Per Million",
            "hasChildren": true,
            "index": 2,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        New Mexico
              </b>
               had the greatest number of police shootings per million residents between 2015 and 2016. Western states in general demonstrated higher rates than the eastern and northern United States, with the 
              <b>
                        District of Columbia
              </b>
               being an exception. Four of the five lowest rates were found in the northeastern United States. 
              <br />
              <br />
              Hover over a state to obtain the number of shootings per million.
    </div>,
            "route": "/percapita",
            "title": "Shootings Per Million By State",
            "type": "map",
          },
          "title": "Shootings Per Million By Race White",
          "type": "map",
        },
        "/shootingsbydate": Object {
          "descTitle": "Total Shootings By Date",
          "hasChildren": true,
          "index": 3,
          "isLastRoute": true,
          "jsx": <div
            className="text"
    >
            Police involved shootings display a more or less stochastic distribution across the time period surveyed. 
            <b>
                    July 2015
            </b>
             and 
            <b>
                    March 2015
            </b>
             had the most shootings of any month at 
            <b>
                    124
            </b>
             and 
            <b>
                    113
            </b>
            , respectively. 
            <b>
                    June 2015
            </b>
             and 
            <b>
                    April 2016
            </b>
             tied for the least, each with 
            <b>
                    80
            </b>
             shootings.
    </div>,
          "title": "Shootings By Date",
          "type": "line",
        },
        "/shootingsbydate/asian": Object {
          "childIndex": 2,
          "descSubtitle": "Asian",
          "descTitle": "Total Shootings By Date",
          "hasNextSibling": true,
          "parent": Object {
            "descTitle": "Total Shootings By Date",
            "hasChildren": true,
            "index": 3,
            "isLastRoute": true,
            "jsx": <div
              className="text"
    >
              Police involved shootings display a more or less stochastic distribution across the time period surveyed. 
              <b>
                        July 2015
              </b>
               and 
              <b>
                        March 2015
              </b>
               had the most shootings of any month at 
              <b>
                        124
              </b>
               and 
              <b>
                        113
              </b>
              , respectively. 
              <b>
                        June 2015
              </b>
               and 
              <b>
                        April 2016
              </b>
               tied for the least, each with 
              <b>
                        80
              </b>
               shootings.
    </div>,
            "route": "/shootingsbydate",
            "title": "Shootings By Date",
            "type": "line",
          },
          "title": "Shootings By Date By Race Asian",
          "type": "line",
        },
        "/shootingsbydate/black": Object {
          "childIndex": 0,
          "descSubtitle": "African American",
          "descTitle": "Total Shootings By Date",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            Shootings of African Americans spiked in 
            <b>
                    March of 2015
            </b>
            , when 37 were killed in a single month. The fewest occurred in the 
            <b>
                    last month of 2016
            </b>
            , when 
            <b>
                    14
            </b>
             people were killed. Unfortunately, killings of African Americans remain fairly steady across the two years, with a median of 
            <b>
                    23.5
            </b>
             people per month.
    </div>,
          "parent": Object {
            "descTitle": "Total Shootings By Date",
            "hasChildren": true,
            "index": 3,
            "isLastRoute": true,
            "jsx": <div
              className="text"
    >
              Police involved shootings display a more or less stochastic distribution across the time period surveyed. 
              <b>
                        July 2015
              </b>
               and 
              <b>
                        March 2015
              </b>
               had the most shootings of any month at 
              <b>
                        124
              </b>
               and 
              <b>
                        113
              </b>
              , respectively. 
              <b>
                        June 2015
              </b>
               and 
              <b>
                        April 2016
              </b>
               tied for the least, each with 
              <b>
                        80
              </b>
               shootings.
    </div>,
            "route": "/shootingsbydate",
            "title": "Shootings By Date",
            "type": "line",
          },
          "title": "Shootings By Date By Race African American",
          "type": "line",
        },
        "/shootingsbydate/latino": Object {
          "childIndex": 1,
          "descSubtitle": "Latino",
          "descTitle": "Total Shootings By Date",
          "hasNextSibling": true,
          "parent": Object {
            "descTitle": "Total Shootings By Date",
            "hasChildren": true,
            "index": 3,
            "isLastRoute": true,
            "jsx": <div
              className="text"
    >
              Police involved shootings display a more or less stochastic distribution across the time period surveyed. 
              <b>
                        July 2015
              </b>
               and 
              <b>
                        March 2015
              </b>
               had the most shootings of any month at 
              <b>
                        124
              </b>
               and 
              <b>
                        113
              </b>
              , respectively. 
              <b>
                        June 2015
              </b>
               and 
              <b>
                        April 2016
              </b>
               tied for the least, each with 
              <b>
                        80
              </b>
               shootings.
    </div>,
            "route": "/shootingsbydate",
            "title": "Shootings By Date",
            "type": "line",
          },
          "title": "Shootings By Date By Race Latino",
          "type": "line",
        },
        "/shootingsbydate/nativeamerican": Object {
          "childIndex": 3,
          "descSubtitle": "Native American",
          "descTitle": "Total Shootings By Date",
          "hasNextSibling": true,
          "parent": Object {
            "descTitle": "Total Shootings By Date",
            "hasChildren": true,
            "index": 3,
            "isLastRoute": true,
            "jsx": <div
              className="text"
    >
              Police involved shootings display a more or less stochastic distribution across the time period surveyed. 
              <b>
                        July 2015
              </b>
               and 
              <b>
                        March 2015
              </b>
               had the most shootings of any month at 
              <b>
                        124
              </b>
               and 
              <b>
                        113
              </b>
              , respectively. 
              <b>
                        June 2015
              </b>
               and 
              <b>
                        April 2016
              </b>
               tied for the least, each with 
              <b>
                        80
              </b>
               shootings.
    </div>,
            "route": "/shootingsbydate",
            "title": "Shootings By Date",
            "type": "line",
          },
          "title": "Shootings By Date By Race Native American",
          "type": "line",
        },
        "/shootingsbydate/white": Object {
          "childIndex": 4,
          "descSubtitle": "White",
          "descTitle": "Total Shootings By Date",
          "hasNextSibling": false,
          "isLastChildRoute": true,
          "parent": Object {
            "descTitle": "Total Shootings By Date",
            "hasChildren": true,
            "index": 3,
            "isLastRoute": true,
            "jsx": <div
              className="text"
    >
              Police involved shootings display a more or less stochastic distribution across the time period surveyed. 
              <b>
                        July 2015
              </b>
               and 
              <b>
                        March 2015
              </b>
               had the most shootings of any month at 
              <b>
                        124
              </b>
               and 
              <b>
                        113
              </b>
              , respectively. 
              <b>
                        June 2015
              </b>
               and 
              <b>
                        April 2016
              </b>
               tied for the least, each with 
              <b>
                        80
              </b>
               shootings.
    </div>,
            "route": "/shootingsbydate",
            "title": "Shootings By Date",
            "type": "line",
          },
          "title": "Shootings By Date By Race White",
          "type": "line",
        },
        "/total-shootings": Object {
          "descSubtitle": "By State",
          "descTitle": "Total Shootings",
          "hasChildren": true,
          "index": 1,
          "jsx": <div
            className="text"
    >
            Of all 50 states, 
            <b>
                    California
            </b>
             had the greatest number of shootings in this time period at 
            <b>
                    372
            </b>
            . 
            <b>
                    Texas
            </b>
             and 
            <b>
                    Florida
            </b>
             had the second and third greatest counts at 
            <b>
                    205
            </b>
             and 
            <b>
                    144
            </b>
            , respectively.
             
            <b>
                    North Dakota
            </b>
             had the fewest shootings of all states with only
             
            <b>
                    2
            </b>
            .
            <br />
            <br />
            Hover over a state to obtain the number of shootings.
    </div>,
          "title": "Total Shootings",
          "type": "map",
        },
        "/total-shootings/asian": Object {
          "childIndex": 2,
          "descSubtitle": "Asian",
          "descTitle": "Total Shootings",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            Shootings of Asian Americans demonstrate a high level of spatial variability, with many states having none at all between 2015 and 2016.
             
            <b>
                    California
            </b>
             once again had the greatest number of shootings at
             
            <b>
                    16
            </b>
            . 
            <b>
                    Hawaii
            </b>
             had the second largest number, 
            <b>
                    6
            </b>
            , likely owing to a larger Asian American population than other states.
            <br />
            <br />
            Hover over a state to obtain the number of shootings.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Total Shootings",
            "hasChildren": true,
            "index": 1,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        California
              </b>
               had the greatest number of shootings in this time period at 
              <b>
                        372
              </b>
              . 
              <b>
                        Texas
              </b>
               and 
              <b>
                        Florida
              </b>
               had the second and third greatest counts at 
              <b>
                        205
              </b>
               and 
              <b>
                        144
              </b>
              , respectively.
               
              <b>
                        North Dakota
              </b>
               had the fewest shootings of all states with only
               
              <b>
                        2
              </b>
              .
              <br />
              <br />
              Hover over a state to obtain the number of shootings.
    </div>,
            "route": "/total-shootings",
            "title": "Total Shootings",
            "type": "map",
          },
          "title": "Total Shootings By Race Asian",
          "type": "map",
        },
        "/total-shootings/black": Object {
          "childIndex": 0,
          "descSubtitle": "African American",
          "descTitle": "Total Shootings",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            Shootings of African Americans by police were heavily concentrated along the Eastern seaboard and in the southern United States. However,
             
            <b>
                    California
            </b>
             had the greatest number of shootings of black Americans at
             
            <b>
                    62
            </b>
            . 
            <b>
                    Florida
            </b>
             and 
            <b>
                    Texas
            </b>
             had the second and third highest totals at 
            <b>
                    50
            </b>
             and 
            <b>
                    41
            </b>
            , respectively. States with low African American populations in the intermountain West and the Northeast, such as
             
            <b>
                    Montana
            </b>
            , 
            <b>
                    Idaho
            </b>
            , 
            <b>
                    Maine
            </b>
            , and 
            <b>
                    Vermont
            </b>
             were among the few with no shootings of black Americans.
            <br />
            <br />
            Hover over a state to obtain the number of shootings.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Total Shootings",
            "hasChildren": true,
            "index": 1,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        California
              </b>
               had the greatest number of shootings in this time period at 
              <b>
                        372
              </b>
              . 
              <b>
                        Texas
              </b>
               and 
              <b>
                        Florida
              </b>
               had the second and third greatest counts at 
              <b>
                        205
              </b>
               and 
              <b>
                        144
              </b>
              , respectively.
               
              <b>
                        North Dakota
              </b>
               had the fewest shootings of all states with only
               
              <b>
                        2
              </b>
              .
              <br />
              <br />
              Hover over a state to obtain the number of shootings.
    </div>,
            "route": "/total-shootings",
            "title": "Total Shootings",
            "type": "map",
          },
          "title": "Total Shootings By Race African American",
          "type": "map",
        },
        "/total-shootings/latino": Object {
          "childIndex": 1,
          "descSubtitle": "Latino",
          "descTitle": "Total Shootings",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            Shootings of Latinos by police were spatially concentrated in the Southwest along the U.S. - Mexico border. 
            <b>
                    California
            </b>
             had the greatest number of shootings at 
            <b>
                    154
            </b>
            , more than double that of the second highest,
             
            <b>
                    Texas
            </b>
            , at 
            <b>
                    67
            </b>
            . 
            <b>
                    Arizona
            </b>
            , 
            <b>
                    New Mexico
            </b>
            , and
             
            <b>
                    Colorado
            </b>
             all demonstrated high counts as well.
            <br />
            <br />
            Hover over a state to obtain the number of shootings.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Total Shootings",
            "hasChildren": true,
            "index": 1,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        California
              </b>
               had the greatest number of shootings in this time period at 
              <b>
                        372
              </b>
              . 
              <b>
                        Texas
              </b>
               and 
              <b>
                        Florida
              </b>
               had the second and third greatest counts at 
              <b>
                        205
              </b>
               and 
              <b>
                        144
              </b>
              , respectively.
               
              <b>
                        North Dakota
              </b>
               had the fewest shootings of all states with only
               
              <b>
                        2
              </b>
              .
              <br />
              <br />
              Hover over a state to obtain the number of shootings.
    </div>,
            "route": "/total-shootings",
            "title": "Total Shootings",
            "type": "map",
          },
          "title": "Total Shootings By Race Latino",
          "type": "map",
        },
        "/total-shootings/nativeamerican": Object {
          "childIndex": 3,
          "descSubtitle": "Native American",
          "descTitle": "Total Shootings",
          "hasNextSibling": true,
          "jsx": <div
            className="text"
    >
            Police involved shootings of Native Americans were mainly concentrated in western states with larger indigenous populations. 
            <b>
                    Arizona
            </b>
             had the greatest number of shootings with 
            <b>
                    9
            </b>
            . 
            <b>
                    Alaksa
            </b>
             and
             
            <b>
                    Washington
            </b>
             had the second and third highest counts, at 
            <b>
                    7
            </b>
             and
             
            <b>
                    6
            </b>
             respectively.
            <br />
            <br />
            Hover over a state to obtain the number of shootings.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Total Shootings",
            "hasChildren": true,
            "index": 1,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        California
              </b>
               had the greatest number of shootings in this time period at 
              <b>
                        372
              </b>
              . 
              <b>
                        Texas
              </b>
               and 
              <b>
                        Florida
              </b>
               had the second and third greatest counts at 
              <b>
                        205
              </b>
               and 
              <b>
                        144
              </b>
              , respectively.
               
              <b>
                        North Dakota
              </b>
               had the fewest shootings of all states with only
               
              <b>
                        2
              </b>
              .
              <br />
              <br />
              Hover over a state to obtain the number of shootings.
    </div>,
            "route": "/total-shootings",
            "title": "Total Shootings",
            "type": "map",
          },
          "title": "Total Shootings By Race Native American",
          "type": "map",
        },
        "/total-shootings/white": Object {
          "childIndex": 4,
          "descSubtitle": "White",
          "descTitle": "Total Shootings",
          "hasNextSibling": false,
          "isLastChildRoute": true,
          "jsx": <div
            className="text"
    >
            The majority of people shot by the police between 2015 and 2016 were white.
             
            <b>
                    California
            </b>
            , 
            <b>
                    Texas
            </b>
            , and 
            <b>
                    Florida
            </b>
             had the greatest number of shootings at 
            <b>
                    125
            </b>
            , 
            <b>
                    92
            </b>
            , and 
            <b>
                    74
            </b>
            , respectively. Open carry states like 
            <b>
                    Arizona
            </b>
             and 
            <b>
                    Oklahoma
            </b>
             also had high counts. A high incidence of shootings is also visible in stretches of Appalachia and the southern United States. 
            <br />
            <br />
            Hover over a state to obtain the number of shootings.
    </div>,
          "parent": Object {
            "descSubtitle": "By State",
            "descTitle": "Total Shootings",
            "hasChildren": true,
            "index": 1,
            "jsx": <div
              className="text"
    >
              Of all 50 states, 
              <b>
                        California
              </b>
               had the greatest number of shootings in this time period at 
              <b>
                        372
              </b>
              . 
              <b>
                        Texas
              </b>
               and 
              <b>
                        Florida
              </b>
               had the second and third greatest counts at 
              <b>
                        205
              </b>
               and 
              <b>
                        144
              </b>
              , respectively.
               
              <b>
                        North Dakota
              </b>
               had the fewest shootings of all states with only
               
              <b>
                        2
              </b>
              .
              <br />
              <br />
              Hover over a state to obtain the number of shootings.
    </div>,
            "route": "/total-shootings",
            "title": "Total Shootings",
            "type": "map",
          },
          "title": "Total Shootings By Race White",
          "type": "map",
        },
      },
      "search": "",
      "state": undefined,
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "matchRoute": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "matchRoute": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
